### ISSUE TITLE
error: 'ttnn.avg_pool2d' op Kernel height 7 is greater than input height 1. This ttnn.avg_pool2d configuration is invalid.

### ISSUE THREAD

**Issue by mmilosevicTT** [tenstorrent/tt-mlir#5785]
### Describe the bug

Several versions of `hrnet` TT-XLA PyTorch model are failing in training with the `ttnn.avg_pool2d` after the fix for batch norm training in https://github.com/tenstorrent/tt-mlir/pull/5287. 

Sample outputs:
- [hrnet1_output.txt](https://github.com/user-attachments/files/23475403/hrnet1_output.txt)
- [hrnet2_output.txt](https://github.com/user-attachments/files/23475642/hrnet2_output.txt)

From both files we extract the same stablehlo op that is problematic:
```
%9 = "stablehlo.reduce_window"(%8, %cst_12) <{window_dimensions = array<i64: 1, 1, 7, 7>}> ({
    ^bb0(%arg40: tensor<bf16> loc("reduce-window.42"), %arg41: tensor<bf16> loc("reduce-window.42")):
      %51 = stablehlo.add %arg40, %arg41 : tensor<bf16> loc(#loc52)
      stablehlo.return %51 : tensor<bf16> loc(#loc)
    }) : (tensor<1x2048x13x13xbf16>, tensor<bf16>) -> tensor<1x2048x7x7xbf16> loc(#loc51)
```

Outputs of some versions of these models are provided above.

### Steps to reproduce the issue

1. Go to TT-XLA
2. Replace NOT_SUPPORTED_SKIP tag with EXPECTED_PASSING in the `tests/runner/test_config/torch/test_config_training_single_device.yaml`
3. Run `pytest tests/runner/test_models.py::test_all_models_torch[<TEST_NAME>]`, or any other of the mentioned models.
4. Full list of models failing with this error is present in `tests/runner/test_config/torch/test_config_training_single_device.yaml`

**Comment by mmilosevicTT** [comment]
@sdjordjevicTT @djstefanovicTT @vladimirjovanovicTT Pinging for visibility. Will extract the graph later today.

**Comment by djstefanovicTT** [comment]
```
%9 = "stablehlo.reduce_window"(%8, %cst_12) <{window_dimensions = array<i64: 1, 1, 7, 7>}> ({
    ^bb0(%arg40: tensor<bf16> loc("reduce-window.42"), %arg41: tensor<bf16> loc("reduce-window.42")):
      %51 = stablehlo.add %arg40, %arg41 : tensor<bf16> loc(#loc52)
      stablehlo.return %51 : tensor<bf16> loc(#loc)
    }) : (tensor<1x2048x13x13xbf16>, tensor<bf16>) -> tensor<1x2048x7x7xbf16> loc(#loc51)
```

This particular operation compiles and executes with no errors. The problematic one is:

```
%1877 = "stablehlo.reduce_window"(%cst, %cst_67) <{window_dimensions = array<i64: 7, 7>}> ({
    ^bb0(%arg827: tensor<bf16> loc("reduce-window.7694"), %arg828: tensor<bf16> loc("reduce-window.7694")):
      %1885 = stablehlo.add %arg827, %arg828 : tensor<bf16> loc(#loc2871)
      stablehlo.return %1885 : tensor<bf16> loc(#loc)
    }) : (tensor<7x7xbf16>, tensor<bf16>) -> tensor<1x1xbf16> loc(#loc2870)
```

**Comment by mmilosevicTT** [comment]
This issue still persists for the models mentioned in the decription. I am reopening it and providing on more example that we have encountered.

- inception/pytorch-inceptionv4-single_device-full-training

The output file is: [avg_pool2d_problem.txt](https://github.com/user-attachments/files/24036425/avg_pool2d_problem.txt)
